packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -858,7 +873,10 @@ class TextSelectionOverlay {
     final TextPosition position = renderObject.getPositionForPoint(
       Offset(details.globalPosition.dx, centerOfLineGlobal),
     );
-    _dragStartSelection ??= _selection;
+    _dragStartSelection =
+        defaultTargetPlatform == TargetPlatform.iOS || defaultTargetPlatform == TargetPlatform.macOS
```

Comment:
Consider adding a quick comment here. It's not immediately obvious to me why we behave differently here based on the platform.

Reply:
I kinda thought the same thing while reviewing 👍